src/v/transform/rpc/service.cc

```diff
@@ -94,45 +94,57 @@ local_service::produce(
 ss::future<transformed_topic_data_result> local_service::produce(
   transformed_topic_data data, model::timeout_clock::duration timeout) {
     auto ktp = model::ktp(data.tp.topic, data.tp.partition);
-    auto shard = _partition_manager->shard_owner(ktp);
+    auto result = co_await produce(ktp, std::move(data.batches), timeout);
+    auto ec = result.has_error() ? result.error() : cluster::errc::success;
+    co_return transformed_topic_data_result(data.tp, ec);
+}
+
+ss::future<result<model::offset, cluster::errc>> local_service::produce(
+  model::any_ntp auto ntp,
```

Comment:
unrelated, but i wonder if we should have a naming convention to visually recognize concepts vs concrete types? not sure actually.

Reply:
The style guide says we should be naming this `model::AnyNtp` or something like that. Should I rename in this PR or another?

Reply:
yeh def not in this PR.